src/azure-cli/azure/cli/command_modules/storage/operations/blob.py

```diff
@@ -451,13 +450,13 @@ def _download_blob(*args, **kwargs):
         destination_path = os.path.join(destination, os.path.normpath(blob_normed))
         destination_folder = os.path.dirname(destination_path)
         # Failed when there is same name for file and folder
-        if os.path.isfile(destination_path) and os.path.exists(destination_folder):
+        if os.path.isfile(destination_path) and os.path.exists(destination_folder) and not overwrite:
             raise FileOperationError("%s already exists in %s. Please rename existing file or choose another "
-                                     "destination folder. ")
+                                     "destination folder. " % (destination_path, destination_folder))
```

Comment:
Seems that the message will be `D:\local\dir\container\dir\blob.txt already exists in D:\local\dir\container\dir`... I would suggest to use 
```suggestion
                                     "destination folder. " % (blob_normed, destination))
```